source/extensions/common/redis/cluster_refresh_manager_impl.cc

```diff
@@ -90,6 +90,11 @@ bool ClusterRefreshManagerImpl::onEvent(const std::string& cluster_name, EventTy
           }
         });
         return true;
+      } else if (info->last_callback_time_ms_.load() != last_callback_time_ms) {
+        // If someone else updated the last callback time, then they will trigger the callback.
+        // During this time we don't want to continue to increment the count, so we enforce the
+        // count is 0
+        *count = 0;
```

Comment:
Based on the comment above, should this be a decrement to undo the increment? Or is that not safe since it might also race?

Reply:
decrement can race as well.  If the decrement is called after count got set to 0 in another thread, this would set the counter to -1.